### PR TITLE
Feat(eos_cli_config_gen): BGP VRF IPv4 RM support

### LINKED ISSUE #1756: BGP VRF Address-Family IPv4 Route-map support
### Enhancement summary

please detail your feature enhancement.

We have requirement to apply a route map in Router BGP XX -> vrf xx -> address-family ipv4 level. The route-map command is supported in the EOS shell but we are not able to apply the same route-map in the AVD.

As a work around, we are using the eos_cli to archive the same function. But it would be great to have the route-map variable available for us.

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

Below is the required command in EOS command lines:
router bgp 65202
  vrf OFFNET-SVR
    address-family ipv4
      **neighbor x.x.x.x route-map ALI-DC-OFFNET out**

### Describe the solution you would like

Below is the EOS_CLI workaround in the AVD:

custom_structured_configuration_router_bgp:
  vrfs:
    OFFNET-SVR:
      neighbors:
        x.x.x.x:
          remote_as: 65011
      address_families:
        ipv4:
          neighbors:
            x.x.x.x:
              activate: true
      **eos_cli: |
        address-family ipv4
          neighbor 172.27.2.210 route-map ALI-DC-OFFNET out**

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by danieltudares**
## Change Summary

Support AVD data model for BGP VRF address-family IPv4 Route-maps

## Related Issue(s)

Fixes #1756 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

Support for knob to apply route maps (in/out) under `router_bgp > vrfs > address_families > neighbor` :

```yaml
router_bgp:
  vrfs:
    <vrf>:
      address_families:
        < address_family >:
          neighbors:
            < neighbor_ip_address >:
              route_map_out: < route-map name >
              route_map_in: < route-map name >
```

## How to test

molecule

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen_v4.0/documentation/devices/router-bgp-v4-evpn.md [hunk 1]
```diff
@@ -315,7 +315,11 @@ router bgp 65101
          neighbor 10.2.3.4 activate
          neighbor 10.2.3.4 route-map RM-10.2.3.4-SET-NEXT-HOP-OUT out
          neighbor 10.2.3.5 activate
+<<<<<<< HEAD
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen_v4.0/documentation/devices/router-bgp-v4-evpn.md]: this looks like a bad rebase / merge.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen_v4.0/intended/configs/router-bgp-v4-evpn.cfg [hunk 2]
```diff
@@ -109,7 +109,11 @@ router bgp 65101
          neighbor 10.2.3.4 activate
          neighbor 10.2.3.4 route-map RM-10.2.3.4-SET-NEXT-HOP-OUT out
          neighbor 10.2.3.5 activate
+<<<<<<< HEAD
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen_v4.0/intended/configs/router-bgp-v4-evpn.cfg]: this looks like a bad rebase / merge.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2 [hunk 3]
```diff
@@ -889,6 +889,18 @@ router bgp {{ router_bgp.as }}
 {%                 if neighbor.activate is arista.avd.defined(true) %}
          neighbor {{ neighbor.ip_address }} activate
 {%                 endif %}
+{%                 if router_bgp.vrfs[vrf].address_families[address_family].neighbors[neighbor].route_map_in is arista.avd.defined %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2]: code is duplicated here. I think your rebase went wrong.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen_v4.0/documentation/devices/router-bgp-v4-evpn.md (hunk 1 above)
Comment (danieltudares) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen_v4.0/documentation/devices/router-bgp-v4-evpn.md]: yeah I saw it. Looks like change #1746 affected this change and I need to make some other changes. I will convert this PR into a draft while I sort this.

Comment (danieltudares) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen_v4.0/documentation/devices/router-bgp-v4-evpn.md]: End up messing up this PR pretty bad. I decided to reopen a clean new one #1763 . Sorry for making you review the same code twice Claus!

**Comment by danieltudares** [comment]
Closing this issue and reopening a new clean one #1763 to solve the merge conflicts and some jinja code improvements.

**Pull request closed without merging**